File: 3158_find_the_xor_of_numbers_which_appear_twice/solution.py
# LeetCode 3158 - Find the XOR of Numbers Which Appear Twice
# https://leetcode.com/problems/find-the-xor-of-numbers-which-appear-twice/

from typing import List


class Solution:
    def duplicateNumbersXOR(self, nums: List[int]) -> int:
        cnt = [0] * 51
        ans = 0
        for x in nums:
            cnt[x] += 1
            if cnt[x] == 2:
                ans ^= x
        return ans

File: 3158_find_the_xor_of_numbers_which_appear_twice/Solution.swift
// LeetCode 3158 - Find the XOR of Numbers Which Appear Twice
// https://leetcode.com/problems/find-the-xor-of-numbers-which-appear-twice/

class Solution {
    func duplicateNumbersXOR(_ nums: [Int]) -> Int {
        var cnt = Array(repeating: 0, count: 51)
        var ans = 0
        for x in nums {
            cnt[x] += 1
            if cnt[x] == 2 { ans ^= x }
        }
        return ans
    }
}
